### DISCUSSION TITLE
specified path differs in case from file name on disk [-Werror,-Wnonportable-include-path] on macOS

### DISCUSSION THREAD

**Discussion by RoyZhang2022** [aws/aws-sdk-cpp#2104] [General]
Hi,

I am using the following steps to build AWS SDK on macOS. However I always see an error "specified path differs in case from file name on disk". Checked the ub_core.cpp and you can see it includes a lot of cpp files with nonportable path. It should be the root cause. But how should I avoid this nonportable path? I did not specify the absolute path and do not know where I can disable it. Please indicate. Thanks in advance!

--my steps
```
git clone --recurse-submodules -b "1.9.342" "https://github.com/aws/aws-sdk-cpp.git"
cd aws-sdk-cpp
mkdir install
mkdir build
cd build
cmake ../ -DCMAKE_INSTALL_PREFIX="../install" -DCMAKE_BUILD_TYPE="Release" -DBUILD_ONLY="core;sts;timestream-query" -DCUSTOM_MEMORY_MANAGEMENT="OFF" -DENABLE_TESTING="OFF" -DBUILD_SHARED_LIBS="OFF"
make -j 4

```

--error
```
/Users/royz/downloads/repos/aws-sdk-cpp/build/aws-cpp-sdk-core/ub_core.cpp:20:10: error: non-portable path to file '<Userss/royz/Downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp>'; specified path differs in case from file name on disk [-Werror,-Wnonportable-include-path]
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp>
```

--/Users/royz/downloads/repos/aws-sdk-cpp/build/aws-cpp-sdk-core/ub_core.cpp
```
// Unity Build generated by CMake
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/auth/AWSCredentialsProvider.cpp>
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/auth/AWSCredentialsProviderChain.cpp>
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/auth/SSOCredentialsProvider.cpp>
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/auth/STSCredentialsProvider.cpp>
#include </Users/royz/downloads/repos/aws-sdk-cpp/aws-cpp-sdk-core/source/auth/bearer-token-provider/DefaultBearerTokenProviderChain.cpp>
......

```

**Comment by RoyZhang2022** [comment]
I have identified it is caused by mac disk is in APFS case-insensitive, it should be case-sensitive to make it work.

**Reply by jmklix** [reply]
I'm glad that you where able to find the cause of the above error. Did something need to be changed with the sdk? or just on your end? to fix the case-insensitivity

**Reply by RoyZhang2022** [reply]
I think SDK does not need to be changed. I just created a new partition with case-sensitive. I think you could update document to reflect it needs to be case-sensitive for mac disk.
